monitoring/perf_flag.cc

```diff
@@ -2,28 +2,23 @@
 
 namespace rocksdb {
 #ifdef ROCKSDB_SUPPORT_THREAD_LOCAL
-__thread uint8_t perf_flags[FLAGS_LEN] = {0};
+__thread PerfFlags perf_flags = {};
 #else
-uint8_t perf_flags[FLAGS_LEN] = {0};
+PerfFlags perf_flags = {};
 #endif
 
-void EnablePerfFlag(uint64_t flag) {
-  if (!CheckPerfFlag(flag)) {
-    // & 0b111 means find the flag location is a alternative way to do mod
-    // operation
-    GET_FLAG(flag) ^= (uint64_t)0b1 << ((uint64_t)flag & (uint64_t)0b111);
+PerfFlags NewPerfFlags(std::initializer_list<PerfFlag> l) {
```

Comment:
Could you `&` them? And implement `+` for `PerfFlags` at rust side.

Reply:
In which use case do you think we need `&` perf flags? Typically, we create some `PerfFlags` presets to use, so I don't know when we need to combine flags.

Reply:
After implementing `Add<PerfFlag, Output = PerfFlags>` for `PerfFlag` and `PerfFlags`, you can composing a flags at rust side instead of creating an array of flags and let RocksDB handle it.

Reply:
Ok, sounds like a good idea.

Reply:
I don't modify this rocksdb PR because the C wrappers are only needed to be implemented in `rust-rocksdb`.

Here, the `NewPerfFlags` is only for convenience in rocksdb tests.

The overloaded operators are implemented in https://github.com/tikv/rust-rocksdb/pull/682. Then, we can create a `PerfFlags` by writing `PerfFlag::A | PerfFlag::B | PerfFlag::C`.

PTAL

Reply:
Awesome